### ISSUE TITLE
Expose metrics reporting information status, notification events and error events to 5GMS-Aware Application in M6 Media Session Handling API

### ISSUE THREAD

**Issue by rjb1000** [5G-MAG/Standards#43]
# Context
The M6 Media Session Handling API exposes status information, notification events and error events originating in the Media Session Handler to M6 API consumers, such as the 5GMS-Aware Application. This information may be notified to the end user or (more likely) reported to the 5GMS Application Provider in (private) M8 application logs as part of day-to-day monitoring of service operation.

# Problem description
Clause 12.2.6 describes Media Session Handler status information, notification events and error events relating to the **consumption reporting** feature.

However, the equivalent clause 12.2.7 for the **QoE metrics reporting** feature is currently empty.

# Suggested solution
Add Media Session Handler status information, notification events and error events relating to the QoE metrics reporting feature to clause 12.2.7. As a baseline proposal:
- **Status information relating to metrics reporting:** The QoE metrics report sent most recently to the 5GMS AF at reference point M5.
- **Notification events relating to metrics reporting:** _AF_METRICS_REPORTING_ACTIVATED_, _AF_METRICS_REPORTING_STOPPED_, _NEW_AF_METRICS_REPORT_.
- **Error events relating to metrics reporting:** _ERROR_AF_METRICS_REPORT_ indicating that an error was encountered when attempting to submit a QoE metrics report to the 5GMS AF.

As well as AF-based QoE metrics reporting, the clause 12.2.7 could also specify equivalent status information, notification events and error events for the RAN-based QoE reporting mechanism.

It is recommended that this addition is made in Rel-16 onwards.

**Comment by gheikkila** [comment]
I guess something similar to 12.2.6 could be specified in 12.2.7, at least for the AF-based reporting. But probably need to be a bit different, as for Metrics Reporting you can have multiple configurations active, while Consumption Reporting only have one instance. So the Status Info (i.e. latest metric report) probably somehow need to also contain the metricsReportingConfigurationId (i.e. it's not enough with just the "latest report", as the report doesn't contain the Id)? A similar issue is there for the RAN_based variant as well (although the report identity is handled a bit differently).

**Comment by rjb1000** [comment]
Good point, @gheikkila. Because of the multiplicity, the status object needs to latch the latest QoE metrics report for each metrics reporting configuration. These could, for example, be stored in an associative array indexed by the `metricsReportingConfigurationId` value.

A small potential fly in the ointment is that the `metricsReportingConfigurationId` value assigned at reference point M1 is not exposed to the Media Session Handler at reference point M5 in TS 26.512 V17.3.0. Rather, _ServiceAccessInformation.clientMetricsReportingConfigurations` is a just an array with a simple numerical index. But may a similar array of anonymous QoE metrics reports (one per configuration) is adequate here in the M6 status object?

**Comment by gheikkila** [comment]
The problem with a simple numerical index (i.e. an anonymous array of objects) is that the "scheme" for a configuration controls how you shall interpret the content. Yes, currently we only have the "3GPP-scheme", but in principle other schemes might follow. Without the metricsReportingConfigurationId the receiver of the status info wouldn't know how to read the data. Unless the simple numerical index in some way can be matched towards each individual metricsReportingConfigurationId? 

But the MSH should/must know the metricsReportingConfigurationId, otherwise I cannot see how things will work? For instance, in 11.4.2 the MSH uses the metricsReportingConfigurationId to create the URL towards the AF...?

**Comment by rjb1000** [comment]
> But the MSH should/must know the `metricsReportingConfigurationId`, otherwise I cannot see how things will work? For instance, in 11.4.2 the MSH uses the `metricsReportingConfigurationId` to create the URL towards the AF...?

That looks like a more serious gap in the current specification, @gheikkila. I have raised #46 to capture it.

**Comment by gheikkila** [comment]
Another possible issue: Even if we report the metricsReportingConfigurationId in the status info (together with the metrics reports), any interpretation of the metrics reports by the 5GMS-aware Application requires that it also has knowledge of the clientMetricsReportingConfigurations (to know the scheme). 

This is not obvious, as all of the configuration-related functionality could in principle be handled by the MSH. I guess we **could** also add a way for the Client to ask for the configurations, but that starts to go into just too much complexity (I think). Maybe we can just assume that whoever will use the status info, also somehow will have access to the clientMetricsReportingConfigurations?

**Comment by gheikkila** [comment]
Contribution to 3GPP sent in [https://www.3gpp.org/ftp/TSG_SA/WG4_CODEC/3GPP_SA4_AHOC_MTGs/SA4_MBS/Docs/S4aI230027.zip](url)

**Comment by rjb1000** [comment]
Contributions to 3GPP **SA4#122** addressing this issue:
* TS 26.512:
  * Rel-16: **CR0031** in [S4-230041](https://www.3gpp.org/ftp/TSG_SA/WG4_CODEC/TSGS4_122_Athens/Docs/S4-230041.zip).
  * Rel-17: **CR0032** in [S4-230042](https://www.3gpp.org/ftp/TSG_SA/WG4_CODEC/TSGS4_122_Athens/Docs/S4-230042.zip).

**Comment by rjb1000** [comment]
Contributions revised and approved at 3GPP **SA4#122** Closing Plenary:
* TS 26.512:
  * Rel-16: **CR0031r1** in [S4-230266](https://www.3gpp.org/ftp/TSG_SA/WG4_CODEC/TSGS4_122_Athens/Docs/S4-230266.zip).
  * Rel-17: **CR0032r1** in [S4-230267](https://www.3gpp.org/ftp/TSG_SA/WG4_CODEC/TSGS4_122_Athens/Docs/S4-230267.zip).

**Comment by rjb1000** [comment]
CR pack [SP-250253](https://portal.3gpp.org/ngppapp/CreateTdoc.aspx?mode=view&contributionUid=SP-230253) approved at SA#99 Plenary meeting in Rotterdam.
